src/api/services/giftItem.service.ts

```diff
@@ -0,0 +1,58 @@
+import publicClient from '../clients/publicClient';
+import type { QueryFunctionContext } from '@tanstack/react-query';
+import type {
+  CategoryCardData,
+  GiftItemData,
+  PageParam,
+  QueryKey,
+  ThemedGiftItemsPage,
+  ThemeInfo,
+} from '../types/giftItem.dto';
+
+export const getCategories = async (): Promise<CategoryCardData[]> => {
+  const { data } = await publicClient.get('/api/themes');
```

Comment:
[변경 요청] api 호출 함수들의 매개변수와 반환 타입을 axios instance의 메서드에 제네릭으로 추가하여 자동으로 타입 추론이 되도록 수정해주세요

```suggestion
export const getCategories = async () => {
  const { data } = await publicClient.get<...>('/api/themes');
```